benchmark/test_biocypher_query_generation.py

```diff
@@ -13,142 +19,284 @@
+    )
 
-    with open(FILE_PATH, "a") as f:
-        f.write(
-            f"{prompt_engine.model_name},entities,{calculate_test_score(score)}\n"
-        )
 
+def test_relationship_selection(
+    prompt_engine, test_data_biocypher_query_generation, result_files
+):
+    subtask = inspect.currentframe().f_code.co_name
 
-def test_relationship_selection(prompt_engine):
-    prompt_engine.question = "Which genes are associated with mucoviscidosis?"
-    prompt_engine.selected_entities = ["Gene", "Disease"]
+    kg_schema_file_name = test_data_biocypher_query_generation[0]
+    prompt = test_data_biocypher_query_generation[1]
+    expected_entities = test_data_biocypher_query_generation[2]
+    expected_relationships = test_data_biocypher_query_generation[3]
+    expected_relationship_labels = test_data_biocypher_query_generation[4]
+
+    prompt_engine = prompt_engine(
+        kg_schema_path=f"./benchmark/data/biocypher_query_generation/{kg_schema_file_name}"
+    )
+
+    if benchmark_already_executed(
+        TASK, subtask, prompt_engine.model_name, result_files
+    ):
+        pytest.skip(
+            f"benchmark {TASK}: {subtask} with {prompt_engine.model_name} already executed"
+        )
+
+    prompt_engine.question = prompt
+    prompt_engine.selected_entities = expected_entities
     success = prompt_engine._select_relationships()
     assert success
 
     score = []
-    score.append(
-        prompt_engine.selected_relationships == ["GeneToPhenotypeAssociation"]
-    )
-    score.append(
-        "PERTURBED" in prompt_engine.selected_relationship_labels.keys()
-    )
-    score.append(
-        "source" in prompt_engine.selected_relationship_labels.get("PERTURBED")
-    )
-    score.append(
-        "target" in prompt_engine.selected_relationship_labels.get("PERTURBED")
-    )
-    score.append(
-        "Disease"
-        in prompt_engine.selected_relationship_labels.get("PERTURBED").get(
-            "source"
+    for expected_relationship in expected_relationships:
+        score.append(
+            expected_relationship in prompt_engine.selected_relationships
         )
-    )
-    score.append(
-        "Protein"
-        in prompt_engine.selected_relationship_labels.get("PERTURBED").get(
-            "target"
+
+    for expected_relationship_label_key in expected_relationship_labels.keys():
+        score.append(
+            expected_relationship_label_key
+            in prompt_engine.selected_relationship_labels.keys()
         )
+
+        for expected_relationship_label_value in expected_relationship_labels[
+            expected_relationship_label_key
+        ]:
+            score.append(
+                expected_relationship_label_value
+                in prompt_engine.selected_relationship_labels[
+                    expected_relationship_label_key
+                ]
+            )
+
+    # TODO: make it more generic to be able to compare arbitrarily nested structures
```

Comment:
There are some improvements to the benchmarking of queries using regex in the PR I just merged into the `benchmark` branch by Yuyao; we should merge these functions here and use them to genericise the score generation. https://github.com/biocypher/biochatter/blob/f970a48ebd967539f19862b78af2a496e1af0356/benchmark/test_biocypher_query_generation.py#L386C8-L386C8